edgeless_node/src/resources/ollama.rs

```diff
@@ -0,0 +1,344 @@
+// SPDX-FileCopyrightText: © 2023 Claudio Cicconetti <c.cicconetti@iit.cnr.it>
+// SPDX-License-Identifier: MIT
+
+use futures::{SinkExt, StreamExt};
+
+#[derive(Clone)]
+pub struct OllamaResourceProvider {
+    inner: std::sync::Arc<tokio::sync::Mutex<OllamaResourceProviderInner>>,
+}
+
+enum OllamaCommand {
+    // model_name, history_id, prompt, resource_id, reply_receiver
```

Comment:
Would put this into a struct or use basic wrapper types such as `struct ModelName(String)`

Reply:
Uhm... I guess this is subjective. I like tuples and I don't mind not having named variables (which is the reason why you are suggesting to have a struct instead).

What do you mean by "use basic wrapper types", I am not sure I understand how to apply it here 🤔

Reply:
I am not a fan of having to look for a comment to understand which string parameter is what.
Thats why I would avoid that by letting the type system provide the documentation.